spack/packages/dla-future/package.py

```diff
@@ -46,6 +46,7 @@ class DlaFuture(CMakePackage, CudaPackage, ROCmPackage):
     )
 
     depends_on("cmake@3.22:", type="build")
+    depends_on("pkgconfig", type="build")
```

Comment:
Shouldn't this be needed only when not using MKL, or am I missing something?

Reply:
No, I think you are right. Is there an easy way to express this with `when=`?